pkg/graveler/graveler.go

```diff
@@ -509,6 +511,13 @@ type VersionController interface {
 	// Note: Ancestors of previously expired commits may still be considered if they can be reached from a non-expired commit.
 	SaveGarbageCollectionCommits(ctx context.Context, repository *RepositoryRecord, previousRunID string) (garbageCollectionRunMetadata *GarbageCollectionRunMetadata, err error)
 
+	// GetGarbageCollectionUncommitted Returns a list of at most GCMaxUncommittedObjects uncommitted objects, and a continuation token in the form of GCUncommittedMark if needed.
+	GetGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, mark GCUncommittedMark) ([]*ValueRecord, *GCUncommittedMark, error)
```

Comment:
Needs an iterator. Passing a 1M objects here with a potential of 1000 bytes per entry is 1GB of memory for a single call.

Reply:
And then the mark is redundant. Simply read from the iterator as long as you want.

Reply:
Agreed that this needs to be an iterator. Also, I wonder if the mark is returned to the client (I assume yes, because it is serialized and encoded) - but wondering about the motivation for requiring the client to make multiple paginated calls here.

If it's because we worry that the request will take too long, then this needs to be an asynchronous request - i.e. return a token with which the client can sample the server until the file is ready.

Reply:
Asynchronous requests are very hard to manage. Need to store the request and manage its state in a multi instances case. We normally push this responsibility to the client, AFAIR we still don't have a single handling of async requests, even in places where it's impossible to paginate. E.g a commit of 1B objects.